### PR TITLE
[tool] Check for `test` and `flutter_test` in non-dev dependencies

### PR DESCRIPTION
**Pull request by ekuleshov**
*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

Additional checks as a followup to https://github.com/flutter/packages/pull/6446

*List which issues are fixed by this PR. You must list at least one issue.*

Related to https://github.com/flutter/flutter/issues/145992

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [ ] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: script/tool/test/pubspec_check_command_test.dart [hunk 1]
```diff
@@ -1743,7 +1743,11 @@ ${_topicsSection()}
         package.pubspecFile.writeAsStringSync('''
 ${_headerSection('a_package')}
 ${_environmentSection()}
-${_dependenciesSection(<String>['integration_test: \n    sdk: flutter'])}
+${_dependenciesSection(<String>[
+              'integration_test: \n    sdk: flutter',
+              'flutter_test: \n    sdk: flutter',
```
Comment (stuartmorgan) [inline script/tool/test/pubspec_check_command_test.dart:1748]: If the existing tests are going to change to test more things, then the test descriptions need to change accordingly.

File: packages/css_colors/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ## NEXT
```
Comment (stuartmorgan) [inline packages/css_colors/CHANGELOG.md:1]: This needs a version change, which is why CI is failing.

File: packages/css_colors/pubspec.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ name: css_colors
 description: Defines constant dart:ui Color objects for CSS colors (for use in Flutter code).
 repository: https://github.com/flutter/packages/tree/main/packages/css_colors
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+css_colors%22
-version: 1.1.4
+version: 1.1.4+1
```
Comment (stuartmorgan) [inline packages/css_colors/pubspec.yaml:5]: Please see the end of https://dart.dev/tools/pub/versioning#semantic-versions. We follow the advice there about when to use (and not use) `+` in versions.

File: packages/web_benchmarks/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 1.2.1+1
```
Comment (stuartmorgan) [inline packages/web_benchmarks/CHANGELOG.md:1]: Same issue here.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks!

File: script/tool/test/pubspec_check_command_test.dart (hunk 1 above)
Comment (ekuleshov) [reply script/tool/test/pubspec_check_command_test.dart:1748]: updated test name to 'fails when integration_test, flutter_test or test are used in non dev dependency'

File: packages/css_colors/CHANGELOG.md (hunk 2 above)
Comment (ekuleshov) [reply packages/css_colors/CHANGELOG.md]: Updated

File: packages/web_benchmarks/CHANGELOG.md (hunk 4 above)
Comment (ekuleshov) [reply packages/web_benchmarks/CHANGELOG.md]: Updated

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
@bparrishMines or @tarrinneal could you do the secondary review here?

**Review by bparrishMines — APPROVED** [review]
LGTM

**Pull request merged**